#!/usr/bin/env python
PACKAGE = "lidar_filter"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


gen.add("fixed_frame", str_t, 0, "Ha megvaltozik az erteke, akkor ujrainditas szukseges.", "left_os1/os1_lidar")

gen.add("topic_name", str_t, 0, "Topic neve, amire fel akarunk iratkozni. Ha megvaltozik az erteke, akkor ujrainditas szukseges.", "/left_os1/os1_cloud_node/points")


gen.add("epsilon", double_t, 0, "Meroleges tavolsagmereskor a kuszobertek.", 0.3, 0.0, 1.0)


gen.add("min_x", double_t, 0, "Vizsgalt terulet X (min) koordinataja meterben.", 0, -50, 50)

gen.add("max_x", double_t, 0, "Vizsgalt terulet X (max) koordinataja meterben.", 30, -50, 50)

gen.add("min_y", double_t, 0, "Vizsgalt terulet Y (min) koordinataja meterben.", -10, -30, 30)

gen.add("max_y", double_t, 0, "Vizsgalt terulet Y (max) koordinataja meterben.", 10, -30, 30)

gen.add("min_z", double_t, 0, "Vizsgalt terulet Z (min) koordinataja meterben.", -3, -10, 10)

gen.add("max_z", double_t, 0, "Vizsgalt terulet Z (max) koordinataja meterben.", -1, -10, 10)


gen.add("interval", double_t, 0, "LIDAR vertikalis szogfelbontasanak intervalluma.", 0.1800, 0.0100, 10)


gen.add("curb_points", int_t, 0, "Jardaszegelyen a pontok becsult szama.", 5, 1, 30)

gen.add("curb_height", double_t, 0, "Jardaszegely becsult minimalis magassaga meterben.", 0.0500, 0.0100, 0.5000)

gen.add("angle_filter1", double_t, 0, "Harom pont altal bezart szog vizsgalata X = 0 ertek mellett.", 150, 0, 180)

gen.add("angle_filter2", double_t, 0, "Ket vektor altal bezart szog vizsgalata Z = 0 ertek mellett.", 140, 0, 180)


gen.add("beam_zone", double_t, 0, "A vizsgalt sugarzona fokban.", 30, 10, 100)


exit(gen.generate(PACKAGE, "lidar_filter", "dynamic_reconf"))